docs/openapi.yaml

```diff
@@ -2851,6 +2854,68 @@ paths:
               example:
                 $ref: ./api/tokens/get-fungible-tokens-metadata-list.example.schema.json
 
+  /extended/v1/tokens/nft/holdings:
+    get:
+      operationId: get_nft_holdings
+      summary: Non-Fungible Token holdings
+      description: |
+        Retrieves the list of Non-Fungible Tokens owned by the given principal (STX address or Smart Contract ID).
+        Results can be filtered by one or more asset identifiers and can include metadata about the transaction that made the principal own each token.
+
+        More information on Non-Fungible Tokens on the Stacks blockchain can be found [here](https://docs.stacks.co/write-smart-contracts/tokens#non-fungible-tokens-nfts).
+      tags:
+        - Non-Fungible Tokens
+      parameters:
+        - name: principal
+          in: query
+          description: token owner's STX address or Smart Contract ID
+          required: true
+          example: SPNWZ5V2TPWGQGVDR6T7B6RQ4XMGZ4PXTEE0VQ0S.marketplace-v3
+          schema:
+            type: string
+        - name: asset_identifiers
+          in: query
+          description: identifiers of the token asset classes to filter for
+          required: false
+          example: SPQZF23W7SEYBFG5JQ496NMY0G7379SRYEDREMSV.Candy::candy
+          schema:
+            type: array
+            items:
+              type: string
+        - name: limit
+          in: query
+          description: max number of tokens to fetch
+          required: false
+          schema:
+            type: integer
+        - name: offset
+          in: query
+          description: index of first tokens to fetch
+          required: false
+          schema:
+            type: integer
+        - name: unanchored
+          in: query
+          description: whether or not to include tokens from unconfirmed transactions
+          required: false
+          schema:
+            type: boolean
+        - name: tx_metadata
```

Comment:
Should this be something like a separate endpoint? Seems difficult to specify the spec/docs when the response payload schema can change based off a query param.

Reply:
You can see the response possibilities in Redoc but the UX is kinda bad because it's behind a couple of clicks (click on Responses > results here: https://deploy-preview-936--stacks-blockchain-api.netlify.app/#operation/get_nft_holdings). It's a good point, though, because I can't set two examples, only one.

@saralab what do you think it's best from a docs perspective? I took this behavior from some other metadata endpoints which can show/hide metadata fields in responses according to a query param, but maybe that's worth revisiting.

Reply:
Actually, scratch that 😎 
OpenAPI does support multiple response examples, check it out: https://deploy-preview-936--stacks-blockchain-api.netlify.app/#operation/get_nft_holdings

Reply:
Here is what the generated client code (`@stacks/blockchain-api-client`) looks like when using multiple response schemas:
![image](https://user-images.githubusercontent.com/1447546/148100901-964e7f4a-91b2-4b39-b949-ca11013fa907.png)

There's a known bug with the generator where it can't create proper types with the complex `transaction` schema, so I tested the same multiple-response with another endpoint and getting the same result of `object`:

![image](https://user-images.githubusercontent.com/1447546/148103226-38066cb2-3759-42f7-a048-ab0b17acc57f.png)

It looks like there's no way in OpenAPI to correlate/specify a query param with a response type:
https://stackoverflow.com/questions/36576447/swagger-specify-two-responses-with-same-code-based-on-optional-parameter

This isn't necessarily a deal breaker, but definitely something to consider.

Reply:
Yeah, I've been looking for a standard and haven't come up with anything solid. It's mostly based on opinion.

That said, this is the closest I found that endorses a dynamic response structure based on params (by Microsoft): https://docs.microsoft.com/en-us/onedrive/developer/rest-api/concepts/optional-query-parameters?view=odsp-graph-online

My opinion is that given that we're already changing responses in other endpoints we could just keep using that same practice in the future.